server/server.manifest.template

```diff
@@ -0,0 +1,108 @@
+
+# General notes:
+# - All mount points are mounted using the default 'chroot' type.
+# - `path`: names of directories and files in Gramine environment; they may be
+#           arbitrary but here we mostly reuse host URIs for simplicity (except
+#           for the first `/lib` mount point).
+# - `uri`:  names of directories and files on the host, somewhat confusingly
+#           prepended by the 'file:' keyword.
+
+fs.mounts = [
+  # Mount on-host directory to Gramine glibc/runtime libraries (in 'uri') into
+  # in-Gramine visible directory /lib (in 'path').
+  { path = "/lib", uri = "file:{{ gramine.runtimedir() }}" },
+  { path = "{{ arch_libdir }}", uri = "file:{{ arch_libdir }}" },
+  { path = "/usr", uri = "file:/usr" },
+  { path = "/etc", uri = "file:/etc" },
+  { path = "/{{ server_bin }}", uri = "file:{{ server_bin }}" },
+]
+
+############################### SGX: GENERAL ##################################
+
+sys.stack.size = "2M"
+sys.enable_extra_runtime_domain_names_conf = true
+
+sgx.nonpie_binary = true
+sgx.enclave_size = "{{ enclave_size }}"
+sgx.edmm_enable = false
+sgx.max_threads = 128
+
+sgx.remote_attestation = "{{ ra_type }}"
+
+sgx.isvprodid = {{ isvprodid }}
+sgx.isvsvn    = {{ isvsvn }}
+
+############################# SGX: TRUSTED FILES ###############################
+
+# Specify all files used by Server and its dependencies (including all libraries
+# which can be loaded at runtime via dlopen), as well as other static read-only
+# files (like configuration files).
+#
+# The paths to files are on-host paths. These files will be searched for in
+# in-Gramine visible paths according to mount points above.
+#
+# As part of the build process, Gramine-SGX script (`gramine-sgx-sign`) finds
+# each specified file, measures its hash, and adds it to the manifest entry for
+# that file (converting each entry to a table with "uri" and "sha256" keys).
+# Note that this happens on the developer machine or a build server. If a
+# directory is specified in the list below, then this directory is recursively
+# traversed and each found file is processed as described above.
+#
+# At runtime, during loading of each "trusted file", Gramine-SGX measures its
+# hash and compares with the "sha256" value in the corresponding manifest entry.
+# If hashes match, this file is trusted and allowed to be loaded and used. Note
+# that this happens on the deployment machine.
+
+sgx.trusted_files = [
+  "file:{{ gramine.libos }}",
+  "file:{{ gramine.runtimedir() }}/",
+  "file:{{ arch_libdir }}/",
+  "file:/usr/lib/ssl/",
+  "file:/usr/{{ arch_libdir }}/",
+  "file:/usr/lib/libdcap_quoteprov.so",
+  "file:/etc/ssl/certs/ca-certificates.crt",
+  "file:{{ server_bin }}",
+]
+
+sgx.allowed_files = [
+  "file:/etc/nsswitch.conf",
+  "file:/etc/localtime",
+  "file:/etc/hosts",
+  "file:/etc/ld.so.cache",
+  "file:/etc/host.conf",
+  "file:/etc/passwd",
+  "file:tls/tlskey.der",
```

Comment:
With this we would be writing the tls key and cert unencrypted to the hostfs. Ideally the tls keys are saved to memory only via gramine tmpfs. The tls ket and cert locations `/tmp/tlskey.der` and  `/tmp/tlscert.der` can be left hardcoded and there's no need to configure them via env vars.

Reply:
Vice versa with `client`